pkg/actuators/machine/reconciler.go

```diff
@@ -92,15 +106,12 @@ func (r *Reconciler) create() error {
 	}
 
 	klog.Infof("Created Machine %v", r.machine.Name)
-	if err = r.setProviderID(instance); err != nil {
-		return fmt.Errorf("failed to update machine object with providerID: %w", err)
-	}
-
-	if err = r.setMachineCloudProviderSpecifics(instance); err != nil {
-		return fmt.Errorf("failed to set machine cloud provider specifics: %w", err)
-	}
-
 	r.machineScope.setProviderStatus(instance, conditionSuccess())
+	// DO NOT set addresses on the first pass.
+	// If we set addresses, the machine controller implies that the machine is provisioned.
+	// We remove them here so that we get a chance to requeue when there is a delay in the
+	// instance appearing within the API.
```

Comment:
Suggest to add this link https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/machine/controller.go#L680 here.